### ISSUE TITLE
Incorporating sample weights in loss function

### ISSUE THREAD

**Issue by marlens123** [qubvel/segmentation_models#578]
hey, thanks for providing this great repository! 
I am working on using the sm.Unet model for semi-supervised segmentation with categorical focal dice loss function. My current goal is to implement sample weights in the loss function to scale the loss depending on the quality of the sample, as is possible in tf loss functions (e.g. https://www.tensorflow.org/api_docs/python/tf/keras/losses/CategoricalCrossentropy). I was wondering if anyone here has worked on this before, and would be happy to exchange ideas!
